### PR TITLE
SSR checksum

### PR DESCRIPTION
**Pull request by jonzee**

### CODE REVIEW

File: server/server-prod.js [hunk 1]
```diff
@@ -13,6 +13,12 @@ function send(res, content, context) {
   });
 }
 
+function errorHandler(err, req, res, next) {
+  console.log('API server error ' + err);
```
Comment (michalmikolajczyk) [inline server/server-prod.js]: rm

File: server/server-prod.js [hunk 2]
```diff
@@ -13,6 +13,12 @@ function send(res, content, context) {
...
+  console.log(res);
```
Comment (michalmikolajczyk) [inline server/server-prod.js]: rm

File: src/ts/Offer/Create/Inputs.tsx [hunk 3]
```diff
@@ -30,6 +30,7 @@ const inputGenerator = (type, key, label) => {
       return (
         <div className="col">
           <FormsyText
+            id={`offer-create-${label}`}
```
Comment (michalmikolajczyk) [inline src/ts/Offer/Create/Inputs.tsx]: I don't really like using `id` tags, do we need them? Also, check out https://www.npmjs.com/package/react-html-id

Comment (jonzee) [inline src/ts/Offer/Create/Inputs.tsx]: Yes, `material ui` generates own `id`'s that's why we had a problem with server side rendering, I described this in issue #57

Comment (jonzee) [inline src/ts/Offer/Create/Inputs.tsx]: I added `react-html-id` to generate own `id`'s

Comment (jonzee) [inline src/ts/Offer/Create/Inputs.tsx]: Update: `react-html-id` doesn't work with server side (it's funny, there are some methods described in readme that aren't implemented and exported), I leave this with hardcoded `id`s.

File: server/server-prod.js [hunk 4]
```diff
@@ -45,22 +48,22 @@ module.exports = {
     app.get('/offer/*', (req, res, next) => {
       const id = req.url.split('/')[2];
       request(API + '/offer/' + id, (err, response, body) => {
-        if (err || response.statusCode != 200) {
-          console.log('API server error ' + err);
-          console.log(response);
-          return next();
-        }
+        if (err || response.statusCode != 200) return errorHandler(err, req, response, next);
```
Comment (michalmikolajczyk) [inline server/server-prod.js]: ładne

File: server/server-prod.js [hunk 5]
```diff
@@ -45,22 +48,22 @@ module.exports = {
...
         render(req.url, data)
           .then(app => send(res, app, JSON.stringify(data)))
           .catch(err => res.send('Internal server error'));
       })
     });
 
     app.get('*', function (req, res) {
-      render(req.url)
-        .then(app => send(res, app))
+      const data = { userAgent: req.headers['user-agent'] }
```
Comment (michalmikolajczyk) [inline server/server-prod.js]: semicolon?

File: server/server-prod.js [hunk 6]
```diff
@@ -45,22 +48,22 @@ module.exports = {
...
+      render(req.url, data)
+        .then(app => send(res, app, JSON.stringify(data)))
```
Comment (michalmikolajczyk) [inline server/server-prod.js]: nice style

**Pull request merged**